### PR TITLE
[Feature] Make url-test check HTTP Status 

### PR DESCRIPTION
**Pull request by Aobasama**
Make url-test check whether the HTTP response status satisfies a given regex. If no regex is given or ".*" is given, the check is passed, so there would be no extra overhead.
This is useful when some proxies are banned by some service providers (e.g., OpenAI), in which cases a normal HTTP response with 4xx status code could be received but actually the service cannot be accessed.

#670

### CODE REVIEW

**Comment by Aobasama** [comment]
An exmaple configuration can be:
``` yaml
proxy-groups:
- interval: 86340
  name: OpenAI
  proxies: []
  type: url-test
  url: https://chat.openai.com/
  status-pattern: 200.*
```

File: adapter/adapter.go
[the remark this answers is not in the document]
Comment (Aobasama) [reply adapter/adapter.go:1]: Thanks. You are right

**Comment by lmmir** [comment]
对于load-balance 策略无效。

**Comment by zhuangshuli** [comment]
你好，我也迫切需要这个功能的支持，能尽快更新到在下一个版本吗？谢谢！

**Pull request closed without merging**